lib/view_model/wallet_list/wallet_list_view_model.dart

```diff
@@ -99,9 +100,29 @@ abstract class WalletListViewModelBase with Store {
     _walletManager.updateWalletGroups();
 
     for (var group in _walletManager.walletGroups) {
-      if (group.wallets.length == 1) {
-        singleWalletsList
-            .add(convertWalletInfoToWalletListItem(group.wallets.first));
+      bool shouldAddToSingleWallets = group.wallets.any((wallet) {
+        // Check for non-BIP39 wallet types
+        bool isNonBIP39Wallet = !isBIP39Wallet(wallet.type);
+
+        // Check for nano derivation type
+        bool isNanoDerivationType = wallet.type == WalletType.nano &&
+            wallet.derivationInfo?.derivationType == DerivationType.nano;
+
+        // Check for electrum derivation type
+        bool isElectrumDerivationType =
+            (wallet.type == WalletType.bitcoin || wallet.type == WalletType.litecoin) &&
+                wallet.derivationInfo?.derivationType == DerivationType.electrum;
+
+        // Exclude if any of these conditions are true
+        return isNonBIP39Wallet || isNanoDerivationType || isElectrumDerivationType;
+      });
+
+      if (shouldAddToSingleWallets) {
+        for (var wallet in group.wallets) {
+          singleWalletsList.add(convertWalletInfoToWalletListItem(wallet));
+        }
```

Comment:
you are adding all the wallets in this group to single wallet list if `any` wallet was wrong?

so if the whole group was correctly grouped, but only 1 wallet was grouped by mistake, this code, will add all wallets in this group to single wallet list, why not just specifically add the odd wallet

i.e 
```Dart
singleWalletsList.addAll(group.wallets.where((wallet) { // your checks }).map((e) => convertWalletInfoToWalletListItem(e)));
```

Reply:
Yeah, that does make sense

I went with the logic that if one is disrupted, there is a high probability that everything in that group is disrupted. But it's better to handle them individually instead of adding them as a group.